### ISSUE TITLE
onLogout should provide connection information

### ISSUE THREAD

**Issue by gsonnenf** [meteor/meteor#7397]
The onLogout hook was added with a comment that it was analogous to the onLogin hook:

> Adds Accounts.onLogout() a hook directly analogous to Accounts.onLogin()

https://github.com/meteor/meteor/blob/devel/History.md#v1331

I've tested it and it doesn't seem to provide session context analogous to the onLogin hooks.

The login hook code is as follows - 'packages/accounts-base/accounts_server.js' ->
line: 165
```
Ap._successfulLogin = function (connection, attempt) {
  this._onLoginHook.each(function (callback) {
    callback(cloneAttemptWithConnection(connection, attempt));
    return true;
  });
};
```
line 355:
`this._successfulLogin(methodInvocation.connection, attempt);`

The logout hook code is as follows - 'packages/accounts-base/accounts_server.js' ->
line 179: 
```
Ap._successfulLogout = function () {
  this._onLogoutHook.each(function (callback) {
    callback(); <---- no context set
    return true;
  });
};
```

line 535:
 ```
 methods.logout = function () {
    var token = accounts._getLoginToken(this.connection.id);
    accounts._setLoginToken(this.userId, this.connection, null);
    if (token && this.userId)
      accounts.destroyToken(this.userId, token);
    this.setUserId(null);
    accounts._successfulLogout(); // <-------------no arguments passed
  };
```

It would be useful to pass a connection context and perhaps also return the the userId of the person who logged out as an argument (it is currently cleared before the log out is called).

**Comment by abernix** [comment]
Thanks for this feature request.  I think this is reasonable, and this will likely be a candidate for `pull-requests-encouraged`, however I'll cite the [feature request guidelines](https://github.com/meteor/meteor/blob/devel/Contributing.md#feature-requests) for the sake of completeness and because there are a lot of other feature requests:

> You'll have the best chance of getting a change into core if you can build consensus in the community for it. Start by creating a well specified feature request as a Github issue.

> Help drive discussion and advocate for your feature on the Github ticket (and perhaps the forums). The higher the demand for the feature and the greater the clarity of it's specification will determine the likelihood of a core contributor prioritizing your feature by flagging it with the pull-requests-encouraged label.

Specifically, try to work up a consensus (perhaps on the forums) of what fields would be most beneficial, etc.  Shouldn't be too much to figure out here, but +1's accepted!

Related to #6889 #7187

**Comment by gsonnenf** [comment]
Sounds good, thanks for the advice. I was under the impression onLogout was suppose to have a connection context, being analogous to onLogin, according to the patch notes and #7187, so I assumed there was already consensus and that the initial pull omitted this on accident.

**Comment by Ragsboss** [comment]
+1

**Comment by Ragsboss** [comment]
I can volunteer to fix this. Please let me know if anyone else is working on it. Otherwise I'll start working on this early next week (7/18)..

**Comment by abernix** [comment]
This issue is up for grabs!  Well written pull-requests will be accepted if they meet the requirements.  Please read [Contributing](https://github.com/meteor/meteor/blob/devel/Contributing.md#feature-requests), include new tests whenever possible, ask if you have implementation questions and reference this issue number when opening a pull request. :smile:

You can run tests locally for the `accounts-base` package by running this command from a `meteor` checkout.

    ./meteor test-packages accounts-base

**Comment by herb-kuta-colortokens** [comment]
+1

**Comment by Ragsboss** [comment]
Please review my pull request for this bug at https://github.com/meteor/meteor/pull/7433

**Comment by laosb** [comment]
Close since the PR was merged.

**Comment by interfaith** [comment]
after logging in with twitter a hook is needed to cleanup before logging out.
does that exist now ?

**Comment by abernix** [comment]
@interfaith It does exist as of 1.4 but apparently the docs for that feature did not get published when 1.4 came out.  I'll see about getting that pushed to 1.4 docs, but meanwhile, you can view the commit here: https://github.com/meteor/docs/commit/9072697b0015c4d77d0addee4ff7bb2e31fdfd6f
